runtime/nvqir/cutensornet/simulator_cutensornet.cpp

```diff
@@ -71,29 +71,25 @@ void SimulatorTensorNetBase::applyGate(const GateApplicationTask &task) {
   const auto &controls = task.controls;
   const auto &targets = task.targets;
   // Cache name lookup key:
-  // <GateName>_<num control>_<Param>
-  const std::string gateKey =
-      task.operationName + "_" + std::to_string(controls.size()) + "_" + [&]() {
-        std::stringstream paramsSs;
-        for (const auto &param : task.parameters) {
-          paramsSs << param << "_";
-        }
-        return paramsSs.str();
-      }();
+  // <GateName>_<Param>
+  const std::string gateKey = task.operationName + "_" + [&]() {
+    std::stringstream paramsSs;
+    for (const auto &param : task.parameters) {
+      paramsSs << param << "_";
+    }
+    return paramsSs.str();
+  }();
   const auto iter = m_gateDeviceMemCache.find(gateKey);
 
   // This is the first time we see this gate, allocate device mem and cache it.
   if (iter == m_gateDeviceMemCache.end()) {
-    void *dMem = allocateGateMatrix(
-        generateFullGateTensor(controls.size(), task.matrix));
+    void *dMem = allocateGateMatrix(task.matrix);
     m_gateDeviceMemCache[gateKey] = dMem;
   }
-  std::vector<int32_t> qubits;
-  for (const auto &qId : controls)
-    qubits.emplace_back(qId);
-  for (const auto &qId : targets)
-    qubits.emplace_back(qId);
-  m_state->applyGate(qubits, m_gateDeviceMemCache[gateKey]);
+  // Type conversion
+  const std::vector<int32_t> ctrlQubits(controls.begin(), controls.end());
+  const std::vector<int32_t> targetQubits(targets.begin(), targets.end());
```

Comment:
```suggestion
  const std::vector<std::int32_t> ctrlQubits(controls.begin(), controls.end());
  const std::vector<std::int32_t> targetQubits(targets.begin(), targets.end());
```
For consistency and safety.